### PR TITLE
feat(mail-connector): Adding Dropdown for Outbound Connector to switch content type between plain text and html

### PR DESCRIPTION
**Pull request by itsnuyen**
## Description
image

Adding a Dropdown to set the content type of an email so that html templates can be used in this email outbound connector.

Also set the right content type in the smtp send mail 

## Checklist

- [ ] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

**Comment by sbuettner** [comment]
@itsnuyen Thanks for your contribution. We will get back to you!

**Comment by sbuettner** [comment]
@itsnuyen Thanks again for the PR. We discussed this internally with product. We would prefer to offer a Dropdown that allows the following options:
* Plaintext
* HTML
* Plaintext + HTML

Would that work for you and if so could you adapt your PR? Thanks!

**Comment by itsnuyen** [comment]
Hi @sbuettner,

I can adapt the PR with the suggested recommendations. 
How should the HTML + Plaintext work?
Does the user enter a HTML template as well as a plain text and both are send out?

**Comment by sbuettner** [comment]
@itsnuyen Yes, exactly. Its common practice to send both in case the users email client is not able to show HTML:
https://stackoverflow.com/questions/3902455/mail-multipart-alternative-vs-multipart-mixed

**Comment by ingorichtsmeier** [comment]
Hi @sbuettner, as Minh is on vacation for a month and the customer request the HTML emails, it would be good to separate the complicated html/text mix from the current pull request to a new issue and resolve it later.

Would it be OK to merge this change now and add a new issue to follow up on the combination of HTML and plain text?

Or find a way, to extract the plain text from the html and add it to the mail body automatically?

**Comment by itsnuyen** [comment]
Hi
I added a part, where the user can choose a Multi line Mixed as the third option. 
I think adding the possibility to use FEEL in the HTML templating would be a separate feature.

File: connectors/email/src/test/java/io/camunda/connector/email/client/jakarta/JakartaExecutorTest.java
Comment (ingorichtsmeier) [inline connectors/email/src/test/java/io/camunda/connector/email/client/jakarta/JakartaExecutorTest.java:103]: How to add the second part (in plain text) to the email body?

**Comment by sbuettner** [comment]
Thanks for your contriburation and considering the feedback @itsnuyen 

@ingorichtsmeier @mathias-vandaele from the Connectors team will do a review, fix and merge any upcoming changes.

**Review by mathias-vandaele — COMMENTED** [review]
There are a few comments, but overall it's a good PR 👍

**Comment by sbuettner** [comment]
@mathias-vandaele As @itsnuyen is on vacation would you take care of doing the requested changes yourself?

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:599]: Lets use `Plaintext`

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:602]: "HTML"

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:605]: "HTML & Plaintext"

File: connectors/email/src/test/java/io/camunda/connector/email/client/jakarta/JakartaExecutorTest.java
Comment (mathias-vandaele) [reply connectors/email/src/test/java/io/camunda/connector/email/client/jakarta/JakartaExecutorTest.java:49]: I believe we cant do it, I am working on making it work now @ingorichtsmeier

**Comment by mathias-vandaele** [comment]
@sbuettner I modified it, it now manages efficiently a mix of both plain text body and html body. I will wait for approval before rebasing as we are having conflict

File: connectors-e2e-test/connectors-e2e-test-mail/src/test/java/io/camunda/connector/e2e/BaseEmailTest.java
Comment (sbuettner) [inline connectors-e2e-test/connectors-e2e-test-mail/src/test/java/io/camunda/connector/e2e/BaseEmailTest.java:72]: Shouldnt we look at the content type when fetching the plain text part? Or is it guaranteed to always be the first one?

**Review by sbuettner — DISMISSED** [review]

**Review by sbuettner — APPROVED** [review]

**Comment by sbuettner** [comment]
@mathias-vandaele Before merging it would be great if you could create a deploy preview branch

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Environment:
  - [ ] SaaS only
  - [ ] SM only
  - [ ] Both
- Connectors version: <TO_BE_REPLACED>

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*

File: connectors/email/element-templates/email-outbound-connector.json
Comment (sbuettner) [inline connectors/email/element-templates/email-outbound-connector.json:678]:
```suggestion
    "tooltip" : "Email HTML content",
```

File: connectors/email/element-templates/email-outbound-connector.json
Comment (sbuettner) [inline connectors/email/element-templates/email-outbound-connector.json:655]:
```suggestion
    "label" : "Email HTML Content",
```

File: connectors/email/element-templates/email-outbound-connector.json
Comment (sbuettner) [inline connectors/email/element-templates/email-outbound-connector.json:651]:
```suggestion
    "tooltip" : "Email content",
```

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:619]:
```suggestion
    "tooltip" : "Email content type",
```

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:656]:
```suggestion
    "tooltip" : "Email content",
```

File: connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json
Comment (sbuettner) [inline connectors/email/element-templates/hybrid/email-outbound-connector-hybrid.json:683]:
```suggestion
    "tooltip" : "Email HTML content",
```

File: connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java:95]:
```suggestion
            tooltip = "Email  content type",
```

File: connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java:105]:
```suggestion
            tooltip = "Email content",
```

File: connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java:115]:
```suggestion
            label = "Email HTML Content",
```

File: connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java
Comment (sbuettner) [inline connectors/email/src/main/java/io/camunda/connector/email/outbound/protocols/actions/SmtpSendEmail.java:119]:
```suggestion
            tooltip = "Email HTML content",
```

**Review by sbuettner — COMMENTED** [review]
Looks good. Some minor phrasing comments

**Review by johnBgood — APPROVED** [review]

**Pull request merged**
